### DISCUSSION TITLE
Capturing all asserts via XunitTestMethodRunner or IMessageSink

### DISCUSSION THREAD

**Discussion by birojnayak** [xunit/xunit#2531] [General]
Hi Folks,
Wanted to check if I am building a custom test runner, is there a way I can capture all assertions (mostly interested in failed assertion) via XunitTestMethodRunner or IMessageSink or can I replace  [extension](https://github.com/xunit/assert.xunit/blob/43e3e208a597bc1a4c77f2665f39f98e7401c157/FailAsserts.cs) ? any example would be helpful ?

**Comment by bradwilson** [comment]
We do not record individual assertions. Failed assertions will show up as failed test messages (specifically [`ITestFailed`](https://github.com/xunit/abstractions.xunit/blob/main/src/xunit.abstractions/Messages/ITestFailed.cs), which extends [`IFailureInformation`](https://github.com/xunit/abstractions.xunit/blob/main/src/xunit.abstractions/Messages/BaseInterfaces/IFailureInformation.cs) to include exception types, stack traces, and messages).

One example of a message sink consuming and using that would be the default runner reporter message sink: https://github.com/xunit/xunit/blob/v2/src/xunit.runner.utility/Reporters/DefaultRunnerReporterWithTypesMessageHandler.cs
